modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/containers/DXPActivationKeysTable/ActionButton/index.js

```diff
@@ -0,0 +1,108 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * The contents of this file are subject to the terms of the Liferay Enterprise
+ * Subscription License ("License"). You may not use this file except in
+ * compliance with the License. You can obtain a copy of the License by
+ * contacting Liferay, Inc. See the License for the specific language governing
+ * permissions and limitations under the License, including but not limited to
+ * distribution rights of the Software.
+ */
+
+import ClayAlert from '@clayui/alert';
+import ClayIcon from '@clayui/icon';
+import {useState} from 'react';
+import {useApplicationProvider} from '../../../../../common/context/AppPropertiesProvider';
+import {getIconSpriteMap} from '../../../../../common/providers/ClayProvider';
+import {getAllActivationKeys} from '../../../../../common/services/liferay/rest/raysource/LicenseKeys';
+import downloadFromBlob from '../../../../../common/utils/downloadFromBlob';
+import ButtonDropDown from '../../../components/ButtonDropDown';
+
+import {
+	ALERT_DOWNLOAD_TYPE,
+	AUTO_CLOSE_DOWNLOAD_ALERT_TIME,
+	EXTENSION_FILE_TYPES,
+	STATUS_CODE,
+} from '../../../utils/constants';
+import {ALERT_ACTIVATION_KEYS_DOWNLOAD_TEXT} from '../utils/constants';
+
+const ActivationKeysActionsButton = ({accountKey, sessionId}) => {
+	const {licenseKeyDownloadURL} = useApplicationProvider();
+	const [active, setActive] = useState(false);
+	const [
+		activationKeysdownloadStatus,
+		setActivationKeysdownloadStatus,
+	] = useState('');
```

Comment:
```suggestion
	const [
		activationKeysDownloadStatus,
		setActivationKeysDownloadStatus,
	] = useState();
```

Reply:
I don't like to start an state with undefined value, I think that it's not recommended. It don't say or show any information about that state and aren't clear about what type that state will be. Stating with a empty string demonstrate that state will be an string during your lifecycle, and it's possible use the false value of empty string on conditionals comparisons.

Reply:
Make sense, let use this rule for our next reviews!